pkg/registry/servicecatalog/server/options.go

```diff
@@ -103,6 +105,40 @@ func (o Options) ResourcePrefix() string {
 	return o.EtcdOptions.RESTOptions.ResourcePrefix
 }
 
+// KeyRootFunc returns the appropriate key root function for the storage type in o.
+// This function produces a path that etcd or TPR storage understands, to the root of the resource
+// by combining the namespace in the context with the given prefix
+func (o Options) KeyRootFunc() func(api.Context) string {
+	prefix := o.ResourcePrefix()
+	sType, err := o.StorageType()
+	if err != nil {
+		return nil
+	}
+	if sType == StorageTypeEtcd {
+		return func(ctx api.Context) string {
+			return registry.NamespaceKeyRootFunc(ctx, prefix)
```

Comment:
Not all types were `registry.NamespaceKeyRootFunc`. Some were `registry.NoNamespaceKeyRootFunc`. I'm actually not sure any more which are where they need to be. We've duplicated this 'is namespaced' information all over the codebase, and I think it's become inconsistent.